src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -388,24 +404,19 @@ uct_ib_mlx5_post_send(uct_ib_mlx5_txwq_t *wq,
     ucs_memory_bus_store_fence();
 
     /* Set up copy pointers */
-    dst = wq->bf->reg.ptr;
+    dst = wq->reg->addr.ptr;
     src = ctrl;
 
     ucs_assert(wqe_size <= UCT_IB_MLX5_BF_REG_SIZE);
     ucs_assert(num_bb <= UCT_IB_MLX5_MAX_BB);
-    if (ucs_likely(wq->bf->enable_bf)) {
-        /* BF copy */
-        for (n = 0; n < num_bb; ++n) {
-            uct_ib_mlx5_bf_copy_bb(dst, src);
-            dst += MLX5_SEND_WQE_BB;
-            src += MLX5_SEND_WQE_BB;
-            if (ucs_unlikely(src == wq->qend)) {
-                src = wq->qstart;
-            }
-        }
+    if (ucs_likely(wq->reg->mode == UCT_IB_MLX5_MMIO_MODE_BF_POST)) {
+        src = uct_ib_mlx5_bf_copy(dst, src, num_bb, wq);
```

Comment:
It will be safer to make ARM uarch check and introduce DMB ST based on uarch.

Reply:
what do you mean by "uarch"? specific platform check?

Reply:
Yes. Thinking about #ifdef for various arm uarch here. In this case maybe it should be part of uct_ib_mlx5_bf_copy() . AKA at the end of the function.

Reply:
all such ifdefs should better be in arch/aarch64/cpu.h